include/NovelRT/Graphics/ImageRect.h

```diff
@@ -0,0 +1,52 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_GRAPHICS_IMAGERECT_H
+#define NOVELRT_GRAPHICS_IMAGERECT_H
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Graphics {
+
+  class ImageRect : public RenderObject {
+    friend class TextRect;
+
+  private:
+    std::string _imageDir;
+    std::string _previousImageDir;
+    Utilities::Lazy<GLuint> _textureId;
+    std::vector<GLfloat> _uvCoordinates;
+    Utilities::Lazy<GLuint> _uvBuffer;
+    Utilities::Lazy<GLuint> _colourTintBuffer;
+    RGBAConfig _colourTint;
+    std::vector<GLfloat> _colourTintData;
+    LoggingService _logger;
```

Comment:
Given the services are meant to be `unique_ptr` and aren't meant to be cached outside a method call; this shouldn't be a field.

Instead we could either:
* Have `NovelRT` be a "singleton" (you can have one active runner at a time; which you retrieve via a `static NovelRunner* getCurrent()` (which you can then get services from)
-or-
* You could have render objects track the parent render service (since they are tied to the service and need to be recreated if the device is lost anyways); in which case you could walk up to the NovelRunner and then get other services.

Reply:
I think this is a flaw with the current API design for the logging service AFAIK.

I'll see what I can come up with.

Reply:
> I think this is a flaw with the current API design for the logging service AFAIK.

Right, but I think it also applies to the other services. Many of them could require recreation under various scenarios and so we probably don't want users caching bits themselves.

Reply:
Wouldn't reworking this be a part of #87 ?

Reply:
Probably; 90% of my comments are things I'm fine with being addressed in a follow up PR. But things that I wanted to call out as I was reviewing the code.

Reply:
Makes sense - I also just wanted to remind that we do have an issue open so that we can refer back to this when working through it 👍

Reply:
So do I rework this now or not? I'm confused.

Reply:
It can be reworked later. It, like many of the the items I called out, are mostly just observations noted while reviewing the code.

They should be resolved eventually (or at least discussed and we reach consensus on how to handle it or not) but don't have to be resolved "now".

Reply:
Propagated to #101 . Resolving.